File: AIO/AIO 2018/[p3] Cloud Cover.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() {
    int N, K;
    cin >> N >> K;
    vector<int> distances(N - 1);
    vector<int> Kdistances(N - K);
    for(int i = 0; i < N - 1; i++) cin >> distances[i];
 
    int current_sum = 0;
    for(int i = 0; i < K; i++) current_sum += distances[i];
    Kdistances[0] = current_sum;
 
    for(int i = 0; i < N - K - 1; i++) {
        current_sum -= distances[i];
        current_sum += distances[K + i];
        Kdistances[i + 1] = current_sum;
    }
 
    cout << *min_element(Kdistances.begin(), Kdistances.end()) << "\n";
}

File: AIO/AIO 2018/[p3] Cloud Cover.py
N, K = map(int, input().split())
distances = [int(input()) for _ in range(N - 1)]
Kdistances = list()
 
currentsum = sum(distances[:K])
Kdistances.append(currentsum)
 
for i in range(N - K - 1):
    currentsum -= distances[i]
    currentsum += distances[K + i]
    Kdistances.append(currentsum)
 
print(min(Kdistances))
